### PR TITLE
Tests

### PR DESCRIPTION
**Pull request by ahives**
Created some tests

### CODE REVIEW

**Comment by phatboyg** [comment]
I'd rather not have Machete.Tests referencing HL7, there is an HL7 tests...

**Comment by ahives** [comment]
Your suggested changes are checked in now. Please merge.

File: src/Machete.HL7.Tests/AdvancedQueryTests.cs [hunk 1]
```diff
@@ -0,0 +1,223 @@
...
+
+
+    [TestFixture]
+    public class AdvancedQueryTests :
+        MacheteHL7TestContext<Version26>
+    {
+        [Test]
+        public void Should_be_able_to_query_blocks()
+        {
+            const string message = @"MSH|^~\&|NextGen|^DOSC|RH|18779|20130405125146269||ORM^O01|1999077678|P|2.3|||AL|AL
```
Comment (phatboyg) [inline src/Machete.HL7.Tests/AdvancedQueryTests.cs]: What is the source of this sample message? It appears to contain data that may be identifiable. Test data should be hand-crafted, original, and not pulled from a production system.

File: src/Machete.HL7Schema/Version26.cs [hunk 2]
```diff
@@ -0,0 +1,7 @@
+﻿namespace Machete.HL7Schema.V26
+{
+    public interface Version26
```
Comment (phatboyg) [inline src/Machete.HL7Schema/Version26.cs]: This should be added to the model generator, and not hard-coded.

File: src/Machete/DateTimeExtensions.cs [hunk 3]
```diff
@@ -0,0 +1,105 @@
+﻿namespace Machete
+{
+    using System;
+
+
+    public static class DateTimeExtensions
```
Comment (phatboyg) [inline src/Machete/DateTimeExtensions.cs]: Extensions to built-in types are discouraged, as they create noise in the namespace. An extension method on the `Value<DateTimeOffset` would be more appropriate.

**Comment by phatboyg** [comment]
Also, if you could squash/rebase into a single commit and update the PR, that would be helpful.

File: src/Machete.HL7.Tests/AdvancedQueryTests.cs (hunk 1 above)
Comment (ahives) [inline src/Machete.HL7.Tests/AdvancedQueryTests.cs]: This is test data. I will make more changes to it.

File: src/Machete/DateTimeExtensions.cs (hunk 3 above)
Comment (ahives) [inline src/Machete/DateTimeExtensions.cs]: go it. will make these change immediately.

File: src/Machete.HL7Schema/Version26.cs (hunk 2 above)
Comment (ahives) [inline src/Machete.HL7Schema/Version26.cs]: makes sense. I even thought this as well while doing this but the model generator is not accessible. Can you please add it and delete that file.

**Pull request closed without merging**